packages/backend/src/server/routers/campaign.ts

```diff
@@ -42,6 +47,28 @@ export const campaignRouter = router({
         if (campaign) {
           throw new Error('CampaignExists')
         }
+      } else if (input.chainId && input.user) {
+        const chain = getChain(input.chainId)
+
+        if (!chain) {
+          throw new Error('MissingChainId')
+        }
+
+        const provider = new providers.JsonRpcProvider(chain.rpcUrls.default)
+
+        if (!chain.campaignFactory) {
+          throw new Error('MissingCampaignFactory')
+        }
+
+        const factory = new Contract(chain.campaignFactory, campaignFactoryAbi, provider)
+
+        const userCampaigns = await factory.getUserCampaigns(input.user)
+
+        console.dir(userCampaigns)
+
+        if (!userCampaigns) {
```

Comment:
this check does not create any security. It essentially checks that the user deployed any campaign, not that this campaign was deployed by the user. You need to check the returned list to contain the campaign address.